docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+                    ]
+                },
+                "rules": {
+                    ...
+                }
+        ...
+}
+```
+
+`options.order`: Required array of strings.
+                 An array of the names of `fields` or `helpers`, in order, that are to be formatted
+                 within this group. It may contain any field, but only helpers of the type "static".
+ 
+### 4.1.3: Row
+
+A helper of type "row" denotes a row in a table. If rows are not in a table, they should be rendered
+as a "group" helper.
+
+```js
+{
+    "schema" : {
+        ...
+        "helpers" : {
+            "nameOfHelper" : {
+                "type": "row",
+                "description": {
+                    ...
+                },
+                "options": {
+                    "order": [
+                        string,
+                        string,
+                        ...
+                    ]
+                },
+                "rules": {
+                    ...
+                }
+        ...
+}
+```
+
+`options.order`: Required array of strings.
+                 An array of the names of `fields` or `helpers`, in order, that are to be
+                 formatted within the row. It may contain any field, but only helpers of the type "static" and "group".
+
+The size of this array must equal the `options.colSize` value of the table to which the row belongs.
+
+### 4.1.4: Table
+
+A helper of type "table" denotes a list of "row" helpers that are displayed together in a tabular
+form.
+
+A table contains only "row" helpers whose `options.order` array must be of the same size.
+
+```js
+{
+    "schema" : {
+        ...
+        "helpers" : {
+            "nameOfHelper" : {
+                "type": "table",
+                "description": {
+                    ...
+                },
+                "options": {
+                    "colSize" : int,
+                    "rowSize" : int,
+                    "order": [
+                        string,
+                        string,
+                        ...
+                    ],
+                    "showColHeaders" : boolean,
+                    "showDesc" : boolean,
+                    "showAsFooter": boolean,
+                },
+                "rules": {
+                    ...
+                }
+        ...
+}
+```
+`options.colSize` : Required int.
+                    The number of columns in the table. 
+
+`options.rowSize` : Required int.
+                    The number of rows in the table.
+
+`options.order`: Required array of strings.
+                 An array of the names of `row` helpers, in order, that are to be formatted
+                 within this table.
+
+The size of this array must equal `options.rowSize`.
+
+At each index (or column) of every "row" element, the `helper` or `field` that is
+referenced inside that index has to have its `description` match. This description
+will be rendered as the column header if `options.showColHeaders` is true.
```

Comment:
Is there going to be some kind of validator to validate rules like this?

Reply:
yes, I would like to create a JSON schema of the spec that can be used to validate all Z-JSONs http://json-schema.org/. just as JSON-LD also has its schema and validator.